src/dmd/backend/cod4.d

```diff
@@ -443,7 +443,7 @@ void cdeq(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
                 postinc = e11.EV.E2.EV.Vint;
                 if (e11.Eoper == OPpostdec)
                     postinc = -postinc;
-                getlvalue(cdb,&cs,e11,RMstore);
+                getlvalue(cdb,&cs,e1,RMstore);
```

Comment:
https://github.com/dlang/dmd/blob/2ede66c6a9c49f9100501833d7d3e359b05ff336/src/dmd/backend/cod4.d#L713 and
https://github.com/dlang/dmd/blob/master/src/dmd/backend/cgxmm.d#L257
 have more or less the same code, should they be changed, too?

Reply:
The first one seems to have a workaround following it: https://github.com/dlang/dmd/blob/5cb179f804695bf5a9513916b3552a24f77c5eab/src/dmd/backend/cod4.d#L715-L716

Blame leads to this commit: <https://github.com/dlang/dmd/commit/521e68c275c22a20e728be1d2a1ae30d44cd93b9>. Unfortunately, there's hardly a hint as to what issue was fixed there. It looks like removing the workaround, and fixing the `getlvalue` call instead, could work. But maybe we should first find/add a test that breaks without the workaround.

Regarding the one in cgxmm.d, I got nothing.

Reply:
> The first one seems to have a workaround following it:

This code triggers the workaround (Win64):
```
// REQUIRES: -O

int toStringz(string s) { return s.length > 0 ? s[0] : 0; }

private void toAStringz(in string[] a, int*az)
{
    foreach (string s; a)
    {
        *az++ = toStringz(s);
    }
}

void main()
{
    string[1] sa = [ "abc" ];
    int[2] tgt = 0x2a;
    toAStringz(sa[], tgt.ptr);
    assert(tgt[0] == 'a');
    assert(tgt[1] == 0x2a);
}
```
It also passes if e11 is replaced by e1 instead of the workaround.

Reply:
> Regarding the one in cgxmm.d, I got nothing.

It is hit by the same code when using `float`, `double` or `__vector(T)` instead of `int`, but doesn't seem to be affected by the issue.

Reply:
I've added a commit that removes the workaround and adds your test.